libs/langchain/langchain/chains/combine_documents/reduce.py

```diff
@@ -2,13 +2,166 @@
 
 from __future__ import annotations
 
-from typing import Any, Callable, List, Optional, Protocol, Tuple
+from functools import partial
+from typing import Any, Callable, Dict, List, Optional, Protocol, Sequence, Tuple, Union
 
 from langchain_core.documents import Document
+from langchain_core.language_models import LanguageModelLike
+from langchain_core.output_parsers import StrOutputParser
+from langchain_core.prompts import BasePromptTemplate, PromptTemplate
 from langchain_core.pydantic_v1 import Extra
+from langchain_core.runnables import (
+    Runnable,
+    RunnableLambda,
+    RunnableParallel,
+    RunnablePassthrough,
+)
 
 from langchain.callbacks.manager import Callbacks
-from langchain.chains.combine_documents.base import BaseCombineDocumentsChain
+from langchain.chains.combine_documents.base import (
+    DEFAULT_DOCUMENT_SEPARATOR,
+    DOCUMENTS_KEY,
+    BaseCombineDocumentsChain,
+    format_document_inputs,
+    format_documents,
+    validate_prompt,
+)
+
+""" --- LCEL Runnable chains --- """
+
+
+def create_collapse_documents_chain(
```

Comment:
confusing that `create_collapse_documents_chain` is in reduce.py. move to new file `collapse.py`?

Reply:
should the legacy collapse functionality also go there (`collapse_docs`, `split_list_of_docs`)?